src/plugins/intel_gpu/tests/functional/single_layer_tests/dynamic/scaled_dot_product_attention.cpp

```diff
@@ -254,7 +255,38 @@ TEST_P(ScaledAttnLayerGPUTest, CompareWithRefs) {
     run();
 }
 
-const std::vector<std::vector<InputShape>> shapes {
+const std::vector<std::vector<InputShape>> dynamic_shapes_3D {
```

Comment:
Can you please update src/plugins/intel_gpu/tests/functional/subgraph_tests/dynamic/kv_cache_sdpa.cpp tests as well to ensure KV-cache processing works as expected for 3D cases?

Reply:
The related subgraph is complex and manay parts of `make_llm_kv_cache_sdpa_pattern` are aligned to 4D shape case, and I found `scales_zp_output_order` is also aligned to 4D shape in kvcache compression pass. Therefore it will take more time to update to kvcache+sdpa for 3D cases.

Reply:
@andrew-k-park, thanks for checking this, then it's okay not to spend much time on it. Maybe we can just disable KVCacheFusion or KVCacheCompression transformations for ranks != 4?

Reply:
yes. applied it as suggested. let me just leave it in backlog to support the corresponding cases in the future